x-pack/plugins/lens/public/xy_visualization_plugin/xy_config_panel.tsx

```diff
@@ -0,0 +1,248 @@
+    label: 'Right',
+    iconType: 'arrowRight',
+  },
+];
+
+export function XYConfigPanel(props: VisualizationProps<State>) {
+  const { state, datasource, setState } = props;
+
+  return (
+    <EuiForm className="lnsConfigPanel">
+      <EuiFormRow label="Chart type">
+        <EuiButtonGroup
+          legend="Chart Type"
+          name="chartType"
+          className="eui-displayInlineBlock"
+          data-test-subj="lnsXY_seriesType"
+          options={chartTypeIcons}
+          idSelected={state.seriesType}
+          onChange={seriesType => {
+            const isHorizontal = seriesType === 'horizontal_bar';
+            setState({
+              ...state,
+              seriesType: seriesType as SeriesType,
+              x: {
+                ...state.x,
+                position: isHorizontal ? Position.Left : Position.Bottom,
+              },
+              y: {
+                ...state.y,
+                position: isHorizontal ? Position.Bottom : Position.Left,
+              },
+            });
+          }}
+          isIconOnly
+        />
+      </EuiFormRow>
+
+      <EuiFormRow label="Title">
+        <EuiFieldText
+          placeholder="Title"
+          data-test-subj="lnsXY_title"
+          value={state.title}
+          onChange={e => setState({ ...state, title: e.target.value })}
+          aria-label="Title"
+        />
+      </EuiFormRow>
+
+      <EuiFormRow>
+        <EuiSwitch
+          label="Show legend"
+          checked={state.legend.isVisible}
+          data-test-subj="lnsXY_legendIsVisible"
+          onChange={() =>
+            setState({ ...state, legend: { ...state.legend, isVisible: !state.legend.isVisible } })
+          }
+        />
+      </EuiFormRow>
+
+      {state.legend.isVisible && (
+        <EuiFormRow label="Legend position">
+          <EuiButtonGroup
+            legend="Legend position"
+            data-test-subj="lnsXY_legendPosition"
+            name="legendPosition"
+            options={positionIcons}
+            idSelected={state.legend.position}
+            onChange={position =>
+              setState({ ...state, legend: { ...state.legend, position: position as Position } })
+            }
+            isIconOnly
+          />
+        </EuiFormRow>
+      )}
+
+      <EuiFormRow label="X Axis">
+        <>
+          <EuiFormRow label="Title">
+            <EuiFieldText
+              placeholder="Title"
+              data-test-subj="lnsXY_xTitle"
+              value={state.x.title}
+              onChange={e => setState({ ...state, x: { ...state.x, title: e.target.value } })}
+              aria-label="Title"
+            />
+          </EuiFormRow>
+
+          <EuiFormRow label="Value">
+            <NativeRenderer
+              data-test-subj="lnsXY_xDimensionPanel"
+              render={datasource.renderDimensionPanel}
+              nativeProps={{
+                columnId: state.x.accessor,
+                filterOperations: () => true,
```

Comment:
Do we want to allow all operations on x? This would include stuff like "average of bytes". I would suggest starting with a  reduced set of known 'good' things (in this case everything bucketed)

Reply:
Currently the datasource only supports raw operations, so there are no bucketed fields. And we probably want to support raw operations in the future. I'm honestly not sure what this filter should look like. I'll add a TODO comment, though, as it is something we'll want to solve.

Reply:
Sounds good to me. Maybe `isBucketed` is not granular enough for this use case

Reply:
For the XY visualization I would expect all operations to be allowed on X, since we support numbers, strings, and booleans whether they are bucketed or not.

Reply:
My point is that we don't want to allow aggregated metrics on the x axis because this will always be a single value.

Things we want on the x axis: Unaggregated (raw) numbers, strings and dates, bucketed numbers, strings and dates

Things we don't want on the x axis: Metric aggregations on numbers, strings and dates.

The meta information `isBucketed` is not sufficient to differentiate these two classes, so we need something else here. I don't have a good proposal yet, maybe wording these things to make clear they are hints by the datasource which don't _have_ to be followed?

```ts
export interface Operation {
  // ...
  usageHints: {
    bucket?: boolean;
    metric?: boolean;
    cardinality?: 'low' | 'medium' | 'high';
    // ...
  }
}
```
A raw document value (or a column from a user specified SQL query) which the datasource doesn't know anything about would simply contain no hints (except for the case we got cardinality informations because the user opened the "metrics" popover in the field list).

But this is clearly out of scope for this PR, if you think we should discuss this further I can branch it out in a separate issue.